src/gui/widgets/sio1.cc

```diff
@@ -0,0 +1,390 @@
+/***************************************************************************
+ *   Copyright (C) 2022 PCSX-Redux authors                                 *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.           *
+ ***************************************************************************/
+
+#include "gui/widgets/sio1.h"
+
+#include "core/sio1.h"
+#include "gui/gui.h"
+#include "imgui.h"
+#include "imgui_stdlib.h"
+
+struct registerDisplay {
+    const char* description;
+    const char* notes;
+};
+
+static const int kStatusEntries = 32;
+static const int kModeEntries = 16;
+static const int kControlEntries = 16;
+
+// Lots of repeating strings. Need to add a value for bits repeated?
+
+static registerDisplay statusDisplay[kStatusEntries] = {
```

Comment:
Ow, those will hurt translation. Making it better is a bit complex however. We'll do that later.